modules/ai/cognitiveservices/main.bicep

```diff
@@ -0,0 +1,175 @@
+@description('The kind of Cognitive Service to create. See: https://learn.microsoft.com/en-us/azure/cognitive-services/create-account-bicep for available kinds.')
+param kind string
+
+@description('Prefix of Resource Name. Not used if name is provided')
+param prefix string = 'cog'
+
+@description('The location into which your Azure resources should be deployed.')
+param location string = resourceGroup().location
+
+@minLength(2)
+@maxLength(64)
+// Must contain only lowercase letters, hyphens and numbers
+// Must contain at least 2 through 64 characters
+// Can't start or end with hyphen
+@description('The name of the Cognitive Service.')
+param name string = take('${prefix}-${kind}-${uniqueString(resourceGroup().id, location)}', 64)
+
+@description('The tags to apply to each resource.')
+param tags object = {}
+
+@description('A custom subdomain to reach the Cognitive Service.')
+param customSubDomainName string = ''
+
+@description('The Public Network Access setting of the Cognitive Service. When false, only requests from Private Endpoints can access it.')
+param publicNetworkAccess bool = true
+
+@description('A list of private endpoints to connect to the Cognitive Service.')
+param privateEndpoints array = []
+
+@description('The name of the SKU.')
+param skuName string = 'F0'
```

Comment:
looks like another place to potentially use allowed annotation/user-defined type.